csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -209,14 +268,13 @@ private void WriteMessage(TextWriter writer, IMessage message)
                 }
             }
             writer.Write("{ ");
-            bool writtenFields = WriteMessageFields(writer, message, false);
-            writer.Write(writtenFields ? " }" : "}");
+            bool nothingWritten = WriteMessageFields(writer, message, true);
+            writer.Write(nothingWritten ? "}" : " }");
         }
 
-        private bool WriteMessageFields(TextWriter writer, IMessage message, bool assumeFirstFieldWritten)
+        private bool WriteMessageFields(TextWriter writer, IMessage message, bool firstField)
```

Comment:
Please add brief documentation to say what the return value is. Looking over it, I don't mind *too* much about the reversal of the meaning, but it's not as natural to return `true` if a method *hasn't* done anything, IMO. Alternatively, maybe keep the reversal of semantics of the parameter, but keep the original return value meaning.